go/providers.rst

```diff
@@ -1,34 +1,172 @@
+| :param:`normal_searchpath`     | :type:`string`                                                  |
++--------------------------------+-----------------------------------------------------------------+
+| The search path entry under which the :param:`normal_library` would be found.                    |
++--------------------------------+-----------------------------------------------------------------+
+| :param:`race_library`          | :type:`File`                                                    |
++--------------------------------+-----------------------------------------------------------------+
+| The archive file representing the library compiled with the race detector enabled.               |
++--------------------------------+-----------------------------------------------------------------+
+| :param:`race_searchpath`       | :type:`string`                                                  |
++--------------------------------+-----------------------------------------------------------------+
+| The search path entry under which the :param:`race_library` would be found.                      |
++--------------------------------+-----------------------------------------------------------------+
+
 
 GoEmbed
 ~~~~~~~
 
-**TODO**: More information
+GoEmbed is a provider designed to be used as the output of anything that provides go code, and an
+input to anything that compiles go code.
+It combines the source with dependancies that source will require.
+
+There are two main uses for this.
+
+#. Recompiling a library with additional sources.
+   go_library_ returns a GoEmbed provider with the transformed sources and deps that it was
+   consuming.
+   go_test_ uses this to recompile the library with additional test files, to build the test
+   version of the library. You can use the same feature to recompile a proto library with
+   additional sources that were not generated by the proto compiler.
+
+#. Providing the dependancies for generated code.
+   If you wanted to use flatbuffers_ in your code, and you had a custom rule that ran the
+   flatbuffers compiler to generate the serialization functions, you might hit the issue that
+   the only thing that knows you depend on ``github.com/google/flatbuffers/go`` is the generated
+   code.
+   You can instead have the generator return a GoEmbed provider instead of just the generated
+   files, allowing you to tie the generated files to the additional dependancies they add to
+   any package trying to compile them.
+
++--------------------------------+-----------------------------------------------------------------+
+| **Name**                       | **Type**                                                        |
++--------------------------------+-----------------------------------------------------------------+
+| :param:`srcs`                   | :type:`depset(File)`                                           |
```

Comment:
This row is rendering as a single column.

Reply:
Fixed